chatmd/local_sync.py

```
@@ -0,0 +1,197 @@
+        wanted_set = set(wanted)
+        for path in compute_prune_paths(output_dir, wanted_set):
+            try:
+                if path.is_dir():
+                    import shutil
+
+                    shutil.rmtree(path)
+                else:
+                    path.unlink()
+                pruned += 1
+            except OSError:
+                continue
```

Ruff (0.15.9)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)